new-frontiers-14/frontier-station-14
# Inertia dampeners

## About the PR

added Inertia dampening states to allow for players to "cruise" while away from the piloting console
added a new station component in shared to allow for the client to be able to identify a station and prevent the user from being able to send an unanchored request for said station.  also use it to intercept and block any request received by the station threw a third party client


as a pilot, on the nav screen i now have an off, dampen, and anchor button
off sets linier and angular dampening to 1
dampen sets it to the configuration in the ship files
anchor sets it to 1 (it takes about 5 seconds to go from top speed to essentially 0)

this will let pilots cruise and not have as many people go to the doctor because they strained their WASD fingers while flying
this will give miners a little bit of safety if while off their ship, their ship is hit by a stray rock.  then if their ship is anchored it will stop its self relatively quickly so as to not be abandon to that rock for the rest of eternity 

## Why / Balance

i think i covered it above, it was changed to save people from finger cramps, and protect miners from those pilots who don't care about safe driving

## How to test

test 1: cant unanchor POI
- put shuttle console on POI's and make sure that the dampener options are not visible

test 2: UI only allows 1 active state
- buy a shuttle, or remove the station component from the POI of the prior test
- re-open the shuttle console
- click the different options, only 1 should be able to be selected at a time

test 3: dampeners off
- using the same shuttle (or toss some thrusters and gyros on your POI if your that dedicated to it)
- set dampers to off
- View Variables the grid server component physics and observe angular and linier dampening = 0
- move your shuttle and let go of controls.  there should be no change in speed once you let go (unless you hit something)

test 4: anchored
- using said shuttle/POI
- set dampeners to anchored
- View the same Variables and observe values now = 1
- test for rapid deceleration by going from top speed to anchored should not instant stop you, but rapidly decelerate

test 5: dampeners on
- using said shuttle or POI (note POI's are going to be a little wonky)
- set dampeners to on
- View the same Variables and observe the values match what is configured in its yaml shuttle component configuration
-- POI's this is normally 999999 
-- Ships this is normally .05
- if on shuttle test for normal deceleration rates we are all use to.  this is the default behavior before any changes were made
- if on POI ensure that this stops at such a speed that you would turn into a puddle on the front wall because going from 20mps to 0 in a millisecond because of 999999 is insane and should crush the vessel but game magics OP 


## Media

![image](https://github.com/new-frontiers-14/frontier-station-14/assets/987235/4d6f695c-7044-4526-95e6-326e61d3a2f2)

- [ X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes


used the power of the partial classes to extract all of the NF related code for this to the NF directory this should greatly ease any future merge conflicts between this code and upstream

**Changelog**



:cl:
- add: Cruise Control
- add: Parking Breaks
- fix: Fixed fun! (your Prada be driftin')
